R/zchunk_L232.industry_USA.R

```diff
@@ -77,6 +107,58 @@ module_gcam.usa_L232.industry_USA <- function(command, ...) {
       select(LEVEL2_DATA_NAMES[["DeleteFinalDemand"]]) ->
       L232.DeleteFinalDemand_USAind  ## OUTPUT
 
+    # The industry_USA_processing function is used in place of a for loop in the old data sytem.
+    # This function checks to see if the input data needs to be expanded to all states or used as
+    # is.
+
+    # industry_USA_processing: is a function that
+    industry_USA_processing <- function(data){
+
+      # Subset the input data frame for the USA region. The subsetted data will be used
+      # to check to see if the data frame needs to be processed, it's assumed that if the USA
+      # is not found in the region column that regions have already been processed.
+
+      check_USA <- filter(data, region == "USA")
+
+      if(nrow(check_USA) == 0){
+
+        # This does not change the entries of the data frame but will strip the attributes
+        # from the input data frame.
+        new_data <- mutate(data, region = region)
+
+      } else {
+
+        # If the input data frame contains USA region information
+        # then expand the input data to all states.
+
+        # Save the column names for the input data frame.
+        df_names <- names(data)
+
+        # Subset for observations in the USA region, expand all input data frame
+        # columns to all USA states.
+
+        data %>%
+          filter(region == "USA") %>%
+          write_to_all_states(names = df_names) ->
```

Comment:
Because I am pretty sure that the tibbles have values for the 32 GCAM regions and need to be filtered before we can write to all states and it doesn't look like the write_to_all_states does that 

``` 
write_to_all_states <- function(data, names) {

  assert_that(is_tibble(data))
  assert_that(is.character(names))

  region <- NULL  # silence package check notes

  if("logit.year.fillout" %in% names) {
    data$logit.year.fillout <- "start-year"
  }

  if("price.exp.year.fillout" %in% names) {
    data$price.exp.year.fillout <- "start-year"
  }

  data %>%
    set_years %>%
    mutate(region = NULL) %>% # remove region column if it exists
    repeat_add_columns(tibble(region = gcamusa.STATES)) %>%
    select(names)
}
```

Reply:
No, sorry, I meant in line 142 why can't we do that? Why define `df_names` above?

Reply:
ohhh I see. There is no need to do the ` write_to_all_states(names = df_names)`  method I like your way better. I can change it.